### DISCUSSION TITLE
Can I use node-dmx over ethernet?

### DISCUSSION THREAD

**Discussion by luccap11** [node-dmx/dmx#108] [General]
Hi, how can I use node-dmx on my LAN?

**Comment by Fensterbank** [comment]
Hi there,
`node-dmx` is technically only a library, while https://github.com/node-dmx/dmx-web is using that library to provide a web interface and a simple web API.

So you could run the `dmx-web` package on your device connected to the DMX controller (e.g. your Raspberry Pi), but you have to configure it the way it listens to Port 80 or 443 (if you go over the internet).
Then inside your ethernet you can call the API to read out things or set DMX channels, e.g. `GET http://192.168.1.15/config` or `POST http://192.168.1.15/state/general`

Take a look at [this code](https://github.com/node-dmx/dmx-web/blob/master/dmx-web.js#L51) to see the possibilities.
The local webinterface started by `dmx-web` does nothing more than calling that API too.

Hope this helps :)
